src/connector/src/sink/file_sink/mod.rs

```diff
@@ -0,0 +1,230 @@
+        // Todo: specify more file suffixes based on encode_type.
+        let suffix = match self.encode_type {
+            SinkEncode::Json => "json",
+            SinkEncode::Parquet => "parquet",
+            _ => unimplemented!(),
+        };
+
+        // Note: sink decoupling is not currently supported, which means that output files will not be batched across checkpoints.
+        // The current implementation writes files every time a checkpoint arrives, so the naming convention is `epoch + executor_id + .suffix`.
+        let object_name = format!(
+            "{}/epoch_{}_executor_{}.{}",
+            self.write_path, epoch, self.executor_id, suffix,
+        );
+        Ok(self
+            .operator
+            .writer_with(&object_name)
+            .concurrent(8)
+            .buffer(SINK_WRITE_BUFFER_SIZE)
+            .await?)
+    }
+
+    async fn create_sink_writer(&mut self, epoch: u64) -> Result<()> {
+        let object_writer = self.create_object_writer(epoch).await?;
+        match self.encode_type {
+            SinkEncode::Parquet => {
+                let props = WriterProperties::builder();
+                self.sink_writer = Some(FileWriterEnum::ParquetFileWriter(
+                    AsyncArrowWriter::try_new(
+                        object_writer,
+                        self.schema.clone(),
+                        SINK_WRITE_BUFFER_SIZE,
+                        Some(props.build()),
+                    )?,
+                ));
+            }
+            SinkEncode::Json => {
+                self.sink_writer = Some(FileWriterEnum::FileWriter(object_writer));
+                unimplemented!();
+            }
+            _ => unimplemented!(),
+        }
+
+        Ok(())
+    }
+
+    async fn append_only(&mut self, chunk: StreamChunk) -> Result<()> {
+        let (mut chunk, ops) = chunk.compact().into_parts();
+        let filters =
+            chunk.visibility() & ops.iter().map(|op| *op == Op::Insert).collect::<Bitmap>();
+        chunk.set_visibility(filters);
+
+        match self
+            .sink_writer
+            .as_mut()
+            .ok_or_else(|| SinkError::Opendal("Sink writer is not created.".to_string()))?
+        {
+            FileWriterEnum::ParquetFileWriter(w) => {
+                let batch = to_record_batch_with_schema(self.schema.clone(), &chunk.compact())?;
+                w.write(&batch).await?;
+            }
+            FileWriterEnum::FileWriter(_w) => unimplemented!(),
+        }
+
+        Ok(())
+    }
+}
+
+fn convert_rw_schema_to_arrow_schema(
+    rw_schema: risingwave_common::catalog::Schema,
+) -> anyhow::Result<arrow_schema::Schema> {
+    let mut schema_fields = HashMap::new();
+    rw_schema.fields.iter().for_each(|field| {
+        let res = schema_fields.insert(&field.name, &field.data_type);
+        // This assert is to make sure there is no duplicate field name in the schema.
```

Comment:
Should we make sure this in `validate`?

Reply:
I think validate is only used to verify whether a connection can be established with the downstream sink. For example, for a file sink, it is used to verify whether the file can be written normally. 
The `rw_schema` here is defined by the user's `create sink` statement and should be verified on the frontend to some extent, while I notice that other sinks also just get the schema directly from param without doing any validation, so I'd like to maintain the current implementation. Feel free to leave your stronger reasons.